File: Android Testing/GoogleAnalytics/app/src/main/java/ru/guar7387/googleanalytics/database/DatabaseProvider.java
package ru.guar7387.googleanalytics.database;

import java.util.List;

import ru.guar7387.googleanalytics.data.TodoItem;

public interface DatabaseProvider {

    public void close();

    public void addTodoItem(TodoItem todoItem);

    public List<TodoItem> getAll();

    public void removeItem(TodoItem todoItem);

}

File: Android Kotlin/KotlinSample/app/src/main/kotlin/ru/guar7387/kotlinsample/database/DatabaseProvider.kt
package ru.guar7387.kotlinsample.database

import ru.guar7387.kotlinsample.data.Earthquake

public trait DatabaseProvider {

    public fun saveEarthquakes(earthquakes: List<Earthquake>,
                               filter: (val earthquake: Earthquake) -> Boolean = {true})

    public fun readEarthquakes(filter: (val earthquake: Earthquake) -> Boolean = {true})
            : List<Earthquake>

    public fun close()
}


